### PR TITLE
Palindrome check

### PR DESCRIPTION
**Pull request by shrutivanw**
CS Fundamentals' assignment to implement a function that checks if the input string is a palindrome and accordingly returns true or false.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,10 @@
...
+# Palindrome Check
+In this assignment, you'll design and implement one of the string manipulation functions that is commonly asked during interviews.
+Remember that a string is an array of characters. Algorithms that worked on restricted arrays will work on strings as well.
+
+## Exercise
+* Design and implement a method that checks if the input string is a palindrome. The method should return true if the input string is a palindrome, and return false if it is not a palindrome.
+**Note**: Palindrome is a word, phrase or sentence that reads the same backwards as it does forwards. e.g. "madam"
+* Share and explain the time and space complexities for your solution.
+* At minimum, your implementation should pass the basic tests.
+**Note**: Do not use Ruby provided functionality for `.reverse` and `.reverse!`. You may use `.length`.
```
Comment (Hamled) [inline README.md]: We should have an extra blank line above this so that the note goes into its own paragraph.

File: specs/palindrome_check_spec.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative '../lib/palindrome_check'
+
+describe "palindrome" do
+  describe "basic tests" do
+    describe "test 1" do
```
Comment (Hamled) [inline specs/palindrome_check_spec.rb]: Same as the other repos, we should probably remove the `describe` blocks that are immediate parents of individual tests.

File: specs/palindrome_check_spec.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        test_string = "empty"
+
+        palindrome_check(test_string).must_equal false
+      end
+    end
+  end
+
+  # if it's a string parameter, check for empty
+  describe "empty string input test" do
+    it "empty string reads the same backwards or forwards" do
```
Comment (Hamled) [inline specs/palindrome_check_spec.rb]: I think this and the nil test should be moved into the edge cases section.

**Review by Hamled — CHANGES_REQUESTED** [review]

File: lib/palindrome_check.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# A method to check if the input string is a palindrome.
+# Return true if the string is a palindrome. Return false otherwise.
+def palindrome_check(my_phrase)
+  puts "NOT IMPLEMENTED"
+  return true
```
Comment (Hamled) [inline lib/palindrome_check.rb]: How would you feel about doing `raise NotImplementedError` here instead?

Best practice I think is generally to avoid `puts` statements and other things that will print to the console for code being tested, because that output kind of gets randomly interspersed with the test runner's output.

**Comment by shrutivanw** [comment]
Thank you, @Hamled PR updated to incorporate feedback.

**Review by Hamled — APPROVED** [review]
